sesame/src/main/scala/org/w3/banana/sesame/SesameTransactor.scala

```diff
@@ -0,0 +1,28 @@
+package org.w3.banana.sesame
+
+import org.openrdf.repository.RepositoryConnection
+import org.w3.banana.{Transactor, SparqlUpdate, RDFStore}
+
+import scala.util.{Failure, Try}
+
+trait SesameTransactor extends Transactor[Sesame,RepositoryConnection] {
+
+  def r[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
+    val result = body
+    conn.close()
+    result
+  }
+
+  def rw[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
```

Comment:
For both `rw` and `r` we have to ensure that the semantics are the same between implementations. For example, if a read transaction A is opened and a value read. If another tx B modifies that value, and A reads that value again, what does it see? There are two cases, btw, that is B commits/not commits.